### ISSUE TITLE
💡[Feature]: Add gradient descent algorithms

### ISSUE THREAD

**Issue by sindhuja184** [recodehive/machine-learning-repos#1604]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

I want to add the types of gradient descent algorithms namely,

1. Batch Gradient Descent
2. Stochastic Gradient Descent
3. Mini- Batch Gradient Descent

### Use Case

1. Training Neural Networks
2. Linear and Logistic regression

### Benefits

Can be applied on large datasets

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I want to work on this issue

**Comment by sanjay-kv** [comment]
see pinned issue
